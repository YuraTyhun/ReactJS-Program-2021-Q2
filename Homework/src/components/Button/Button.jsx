import React from 'react';
import PropTypes from 'prop-types';

const Button = ({title, className, children, onClick}) => (
    <button onClick={onClick} className={className}>{title}{children}</button>
);

Button.propTypes = {
    className: PropTypes.string,
    title: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func
};

Button.defaultProps = {
  className: '',
  title: '',
  children: '',
  onClick: () => {}
};

export default Button;