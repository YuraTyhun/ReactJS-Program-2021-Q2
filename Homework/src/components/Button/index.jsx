import React from 'react';
import PropTypes from 'prop-types';

const Button = ({title, className, children, onClick, type, disabled}) => (
    <button 
      onClick={onClick} 
      type={type} 
      className={className} 
      disabled={disabled}
    >
      {title}{children}
    </button>
);

Button.propTypes = {
    className: PropTypes.string,
    title: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func,
    type: PropTypes.string,
    disabled: PropTypes.bool
};

Button.defaultProps = {
  className: '',
  title: '',
  children: '',
  onClick: () => {},
  type: 'button',
  disabled: false
};

export default Button;