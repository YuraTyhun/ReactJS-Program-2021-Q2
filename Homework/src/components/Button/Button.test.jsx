import React from 'react';
import renderer from 'react-test-renderer';

import Button from './index';

it('Button renders correctly', () => {
  const button = renderer
    .create(<Button title="Save" className="btn-test" iconSrc="''" iconAlt="Close" />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
