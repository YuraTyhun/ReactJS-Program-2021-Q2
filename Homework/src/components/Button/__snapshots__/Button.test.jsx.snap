// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="btn-test"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Save
  <img
    alt="Close"
    src="''"
  />
</button>
`;
